Automattic/wp-calypso
# A4A > Team: Implement team member access control

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/962
Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1039

## Proposed Changes

This PR:

- Implements WP-Admin access control for team members.
- Updates the permissions to support the regex expressions for dynamic paths.

* To allow team members to access the enabled paths
* To show that they need WP-Admin access to view a few features (Boost, Backup, Scan, Monitor, Plugins, Stats & Activity) on the site overview pane.

## Testing Instructions

* Switch to this branch locally and start the server.
* Login as an agency and invite a new member.
* Accept the invitation as a team member and log in using an incognito window. 
* Now refresh the page where the agency owner is logged in and verify you can access all the routes as expected and that nothing has changed by comparing it with the production version.
* Refresh the page where the agency team member is logged in and verify you can access only the following routes and only the allowed menu items are shown as per this issue: https://github.com/Automattic/automattic-for-agencies-dev/issues/955
* Open any site details as an agency team member -> Verify that you can see the WP-Admin access notice as shown below for the following tabs, and you can view the page without being redirected: Boost, Backup, Scan, Monitor, Plugins, Stats & Activity
* Add the team member to the site and verify they can no longer see the notice (Visit {siteURL}/wp-admin/users.php -> Add the team member -> Accept the invitation as a team member)
* Go to Marketplace -> Visit all the tabs on the hosting and products page and you can view it without being redirected. 

Screenshots:

<img width="1727" alt="Screenshot 2024-09-02 at 5 32 42 PM" src="https://github.com/user-attachments/assets/85fd15ff-1980-4bdf-a314-82e50a79ffa1">
<img width="368" alt="Screenshot 2024-09-02 at 5 33 26 PM" src="https://github.com/user-attachments/assets/5d8c2530-0a24-48ed-8058-5e155d6174b9">

NOTE: The padding on the overview itself should be fixed here:

<img width="767" alt="Screenshot 2024-09-02 at 5 34 03 PM" src="https://github.com/user-attachments/assets/58961c21-61ba-4e94-928c-c26644bd30f7">


## Pre-merge Checklist
